### DISCUSSION TITLE
Theme-UI's `sx` prop vs styled-system's API/style props

### DISCUSSION THREAD

**Discussion by sheedy** [system-ui/theme-ui#1156] [General] [answered]
Hey everyone, huge fan of Styled-System and Theme-UI here.

I'm working on a design system with Theme-UI as the base instead of Styled System on the basis of this quote "[The next evolution of Styled System](https://github.com/system-ui/theme-ui) - if this not accurate, someone please set me straight 😄.

I have three closely related questions about style/sx props:

1. Is Theme-UI's `sx` prop intended to replace Styled-System's more expansive [style props](https://github.com/styled-system/styled-system/blob/master/docs/api.md#style)?
2. Why are `sx`'s [margin and padding props](https://theme-ui.com/sx-prop#margin-and-padding) also available as their own [individual props](https://theme-ui.com/components#style-props) on components?
3. What can Styled-System do that Theme-UI can't (and therefore need to be used in addition to Theme-UI)?

---

As far as I can tell, Theme-UI mixes the terms API and props and has:

- A [low-level API](https://theme-ui.com/api)
- A limited set of [style props/API](https://theme-ui.com/components#style-props) for **components**
- The [`sx` prop](https://theme-ui.com/components#props) for anything not covered by the former.

Styled-System has:

- An [API](https://styled-system.com/api) or "style props"

**Comment by hasparus** [comment, marked as answer]
I'm not sure if it will answer your questions, but you might like it: https://jxnblk.com/blog/two-steps-forward/

**Reply by sheedy** [reply]
Thanks @hasparus. I had read that (and literally everything else @jxnblk has written) a couple of weeks ago but just started using it now and forgot about it.

It answers question 1. 2 isn't overly important but still curious. Would love to know the answer to question 3 though as it has the potential to trip me up.

**Reply by hasparus** [reply]
With regards to (3) I'm using both styled-system and Theme UI (in different projects) and I'm not sure if there's anything missing in Theme UI. The API differs, but the capabilities should be more or less the same.

**Reply by jxnblk** [reply]
I think this *might* answer some of #3 https://theme-ui.com/guides/motivation -- i.e. Theme UI is intended to be more "batteries-included" whereas Styled System can be used in various ways and isn't coupled to React or a particular CSS-in-JS library. As for question #2, why not? It's only there as a convenience, but I'm not sold on the idea that the margin/padding props need to be in there

**Reply by sheedy** [reply]
Thanks @jxnblk! Want to make sure I'm using the one will get the most attention from the maintainers. I'm using React and not tied to any particular CSS-in-JS library.
I quite like Styled-System's more expansive props so I'll re-phrase question 2 – why are margin and padding available as their own props _but not all the others (from Styled-System)_?
